app/models/spree/product_decorator.rb

```diff
@@ -0,0 +1,41 @@
+module Spree
+  module ProductDecorator
```

Comment:
Is this something we want to decorate onto the Product class? Or should we build some kind of factory class that encapsulates all the magic?